core/src/main/java/org/elasticsearch/indices/cache/request/IndicesRequestCache.java

```diff
@@ -207,11 +207,6 @@ public void onRemoval(RemovalNotification<Key, Value> notification) {
      * Can the shard request be cached at all?
      */
     public boolean canCache(ShardSearchRequest request, SearchContext context) {
-        // TODO: for now, template is not supported, though we could use the generated bytes as the key
-        if (hasLength(request.templateSource())) {
-            return false;
-        }
```

Comment:
@colings86 that's a leftover from initial cleanups can you bring this back by checking if the request had a template?